### DISCUSSION TITLE
Add `from_look_at` and `from_look_to` to `Quat`

### DISCUSSION THREAD

**Discussion by Lubba-64** [bitshifter/glam-rs#561] [Ideas]
The proposed patch to `Quat`'s impl:
```rust
    /// Creates a [`Quat`] that points towards the `target` position,
    fn from_look_at(position: Vec3, target: Vec3, up: impl TryInto<Dir3>) -> Quat {
        Self::from_look_to(target - position, up)
    }

    /// Creates a [`Quat`] that has a rotation that points in the given `direction`.
    fn from_look_to(direction: impl TryInto<Dir3>, up: impl TryInto<Dir3>) -> Quat {
        let back = -direction.try_into().unwrap_or(Dir3::NEG_Z);
        let up = up.try_into().unwrap_or(Dir3::Y);
        let right = up
            .cross(back.into())
            .try_normalize()
            .unwrap_or_else(|| up.any_orthonormal_vector());
        let up = back.cross(right);
        Quat::from_mat3(&Mat3::from_cols(right, up, back.into()))
    }
```

The bevy engine has methods similar to this on transform, but I needed them on `Quat` directly and it seemed like convenience methods for creating `Quat`s is welcome, so let me know if I can make a PR for this.

**Comment by Lubba-64** [comment]
I will upstream the changes in bevy and replace their impls with these if this gets accepted.

**Comment by Lubba-64** [comment]
Dir3 is not in this lib, but assuming we convert those to Vec3's and normalize them, or not normalize them 🤷 

Not sure what the best solution would be but I'm going to leave this open until I have some feedback

**Comment by bitshifter** [comment]
you are in luck, I already have  a draft PR for this that I was thinking about merging https://github.com/bitshifter/glam-rs/pull/551

bevy assumes a right handed coordinate system, glam provides left and right handed versions of these methods.

**Comment by bitshifter** [comment]
Merged in #551
